shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionFeature.kt

```diff
@@ -0,0 +1,82 @@
+package org.hyperskill.app.step_completion.presentation
+
+import org.hyperskill.app.analytic.domain.model.AnalyticEvent
+import org.hyperskill.app.step.domain.model.Step
+import org.hyperskill.app.step.domain.model.StepRoute
+
+interface StepCompletionFeature {
+    companion object {
+        fun createState(step: Step, stepRoute: StepRoute): State =
+            State(
+                currentStep = step,
+                continueButtonAction = if (stepRoute is StepRoute.Learn) {
+                    ContinueButtonAction.FetchNextStepQuiz
+                } else {
+                    ContinueButtonAction.NavigateToBack
+                }
+            )
+    }
+
+    data class State(
+        val currentStep: Step,
+        val continueButtonAction: ContinueButtonAction,
+        val isPracticingLoading: Boolean = false
```

Comment:
It's used for disabling Start practicing/Continue buttons. We should disable it after `Message.StepSolved` because we need check topic completion to decide where the button should navigate user to, homescreen or back